### PR TITLE
(5.x) Merge 4.x

### PR DESCRIPTION
**Pull request by asmorkalov**
Contrib: https://github.com/opencv/opencv_contrib/pull/3783
Extra: https://github.com/opencv/opencv_extra/pull/1205

#23279 from fengyuentau:add_topk
#25832 from chachoi-world:4.x
#25959 from arvind-iyer:4.x
#25968 from FantasqueX:correct-bayer2gray-simd-1
#25970 from savuor:rv/hal_pyrdown
#25980 from ericmariasis:eric-mariasis-stitcher-py-markdown
#25981 from Kumataro:fix25971
#25986 from asmorkalov:as/js_for_contrib
#25996 from asmorkalov:as/fillPoly_check
#26002 from nishanthdass:doc/missing-fields-python-tutorials
#26004 from ericmariasis:eric-mariasis-issue-26000
#26009 from alexlyulkov:al/unify-build-gradle
#26014 from MaximSmolskiy:increase-search-radius-for-corners-neighbors-in-ChessBoardDetector-findQuadNeighbors
#26018 from Kumataro:fix26017
#26019 from shengyu7697:fix-typo
#26022 from Kumataro:fix26016
#26029 from fengyuentau:dnn/fix_cuda_matmul_crash
#26031 from FantasqueX:fix-typo-2
#26032 from FantasqueX:fix-typo-3
#26034 from Octopus136:issue-26033
#26038 from shengyu7697:fix-typo
#26041 from knmcguire:patch-1
#26045 from asmorkalov:as/variant-free
#26052 from savuor:rv/rounding_lock
#26053 from alexlyulkov:al/opencl-conformance-tests
#26054 from asmorkalov:as/warping_perf
#26059 from Abdurrahheem:ash/fix-einsum-allocation
#26061 from penghuiho:fix-pow-bug
#26062 from asmorkalov:as/pow_ocl_duplicate
#26063 from mshabunin:fix-config-and-doc
#26069 from hanliutong:fix-medianBlur

Ported to opencv_contrib with related PR:
#26046 from ivafanas:speedup_random_forest_get_votes

Previous "Merge 4.x": #25998

### CODE REVIEW

File: modules/dnn/test/test_onnx_importer.cpp [hunk 1]
```diff
@@ -3274,6 +3278,39 @@ TEST_P(Test_ONNX_layers, ClipDivSharedConstant) {
...
+        net.setPreferableTarget(target);
+
+        net.setInput(input);
+        std::vector<Mat> outputs;
+        net.forward(outputs, std::vector<std::string>{"values", "indices"});
+
+        Mat output_res_val = outputs.front(),
+            output_res_ind = outputs.back();
+
+        output_ref_ind.convertTo(output_ref_ind, CV_32F); // TODO: revise this conversion in 5.x
```
Comment (asmorkalov) [inline modules/dnn/test/test_onnx_importer.cpp:3301]: @fengyuentau Could you take a look on TopK behavior in 5.x? There was TODO to remove the conversion. I removed the original one, by have to add another one to fix test. Both `output_ref_ind` and `output_res_ind` are FP32 now.

**Review by mshabunin — APPROVED** [review]

File: modules/dnn/test/test_onnx_importer.cpp (hunk 1 above)
Comment (asmorkalov) [reply modules/dnn/test/test_onnx_importer.cpp:3301]: Filed ticker about it: https://github.com/opencv/opencv/issues/26076

File: modules/core/test/test_operations.cpp [hunk 2]
```diff
@@ -1087,7 +1087,7 @@ bool CV_OperationsTest::operations1()
         Size sz(10, 20);
         if (sz.area() != 200) throw test_excep();
         if (sz.width != 10 || sz.height != 20) throw test_excep();
-        if (cvSize(sz).width != 10 || cvSize(sz).height != 20) throw test_excep();
+        if (sz.width != 10 || sz.height != 20) throw test_excep();
```
Comment (opencv-alalek) [inline modules/core/test/test_operations.cpp:1090]: double check for what?

File: modules/dnn/src/layers/einsum_layer.cpp [hunk 3]
```diff
@@ -541,7 +541,7 @@ class LayerEinsumImpl CV_FINAL : public EinsumLayer
                 // Use either the preprocessed inputs (if it is available) or the corresponding raw inputs
                 result = pairwiseOperandProcess(!result.empty() ? result : rawInputs[0],
                                                 !result.empty() ? tmpResult : homogenizedInputDims[0],
-                                                (!preProcessedInputs.empty() && !preProcessedInputs[input].empty()) ? preProcessedInputs[input] : rawInputs[input],
+                                                (!preProcessedInputs[input].empty()) ? preProcessedInputs[input] : rawInputs[input],
```
Comment (opencv-alalek) [inline modules/dnn/src/layers/einsum_layer.cpp:544]: Why is pre-check removed?

(anyway it is wrong as it doesn't check vector size vs index)

File: modules/dnn/src/layers/einsum_layer.cpp [hunk 4]
```diff
@@ -541,7 +541,7 @@ class LayerEinsumImpl CV_FINAL : public EinsumLayer
                 // Use either the preprocessed inputs (if it is available) or the corresponding raw inputs
```
Comment (opencv-alalek) [inline modules/dnn/src/layers/einsum_layer.cpp:541]: Lost assertion: https://github.com/opencv/opencv/blame/4.x/modules/dnn/src/layers/einsum_layer.cpp#L462

File: modules/dnn/src/layers/einsum_layer.cpp (hunk 3 above)
Comment (asmorkalov) [reply modules/dnn/src/layers/einsum_layer.cpp:545]: After https://github.com/opencv/opencv/pull/26059 the `preProcessedInputs` is resized, but not reserved. As soon as the layer has inputs, the vector is not empty. I just selected 4.x version.

**Review by opencv-alalek — APPROVED** [review]

**Pull request merged**